term_posix.go

```diff
@@ -83,6 +84,42 @@ func (t *Term) setSpeed(baud int) error {
 	return termios.Tcsetattr(uintptr(t.fd), termios.TCSANOW, (*syscall.Termios)(&a))
 }
 
+func clamp(v, lo, hi int64) int64 {
+	if v < lo {
+		return lo
+	}
+	if v > hi {
+		return hi
+	}
+	return v
+}
+
+// timeoutVals converts d into values suitable for termios VMIN and VTIME ctrl chars
+func timeoutVals(d time.Duration) (vmin, vtime uint8) {
+	if d > 0 {
+		// VTIME is expressed in terms of deciseconds
+		vtimeDeci := d.Nanoseconds() / 1e6 / 100
+		// ensure valid range
+		vtime = uint8(clamp(vtimeDeci, 1, 0xff))
+	} else {
+		// block indefinitely until we receive at least 1 byte
+		vmin = 1
+	}
+	return
+}
+
+// SetReadTimeout sets the read timeout, after which io.EOF is returned.
+//
+// A zero value for d means read operations will not time out.
```

Comment:
Can you please implement this following the pattern of SetSpeed above. That is a function ReadTimeout, that is applied with SetOption, and a helper, SetReadTimeout